test_dev.sh

```diff
@@ -0,0 +1,136 @@
+      echo "    integration    Integration tests subcommand, run integration -h for more information"
+      exit 0
+      ;;
+    a) # -a (all)
+      trap "outcome_and_exit" EXIT
+      docker-compose run --use-aliases boulder ./test.sh
+      ;;
+    *) # catch invalid options
+      echo "Invalid Option: $OPTARG use: -h for help" 1>&2
+      exit 1
+      ;;
+  esac
+done
+
+if [ $# -gt 1 ] # Check for subcommands
+then
+  shift $((OPTIND -1))
+  subcommand="${1}"; shift
+
+  case "$subcommand" in
+  unit) # Parse options to the unit sub command
+    while getopts ":d:ha" opt; do
+      case ${opt} in
+        h) # -h (help)
+          echo
+          echo "Usage:"
+          echo "    -h                Displays this help message"
+          echo "    -a                Run all unit tests"
+          echo "    -d <DIRECTORY>    Run unit tests for a specific directory"
+          exit 0
+          ;;
+        a)  # -a (all)
+          trap "outcome_and_exit" EXIT
+          docker-compose run -e RUN="unit" --use-aliases boulder ./test.sh
+          ;;
+        d) # -d (directory)
+          trap "outcome_and_exit" EXIT
+          docker-compose run --use-aliases boulder go test $OPTARG
+          ;;
+        :) # assigns $OPTARG for -d
+          echo "Invalid Option: $OPTARG requires an argument, use: -h for help" 1>&2
+          exit 1
+          ;;
+        *) # catch invalid options
+          echo "Invalid Option: $OPTARG use: -h for help" 1>&2
+          exit 1
+          ;;
+      esac
+    done
+    shift $((OPTIND -1))
+    ;;
+esac
+
+case "$subcommand" in
+  integration) # Parse options to the integration sub command
+    while getopts ":f:ha" opt; do
+      case ${opt} in
+        h) # -h (help)
+          echo
+          echo "Usage:"
+          echo "    -h                   Displays this help message"
+          echo "    -a                   Run all integration tests"
+          echo "    -f <FILTER_REGEX>    Run only those tests and examples matching the regular expression."
+          echo "                         For tests, the regular expression is split by unbracketed slash (/)"
+          echo "                         characters into a sequence of regular expressions, and each part"
+          echo "                         of a test's identifier must match the corresponding element in"
+          echo "                         the sequence, if any. Note that possible parents of matches are"
+          echo "                         run too, so that -f X/Y matches and runs and reports the result"
+          echo "                         of all tests matching X, even those without sub-tests matching Y,"
+          echo "                         because it must run them to look for those sub-tests. Note: this"
+          echo "                         this option disables the '"back in time"' integration test"
+          echo
+          echo "List of integration tests:"
```

Comment:
Personally, I would have the list be a separate `-l` (list) flag, rather than bundled with `-h`, but definitely just a matter of opinion.